sql/core/src/main/scala/org/apache/spark/sql/json/JSONRelation.scala

```diff
@@ -105,6 +107,15 @@ private[sql] class JSONRelation(
     jsonSchema
   }
 
+  override private[sql] def buildScan(
+      requiredColumns: Array[String],
+      filters: Array[Filter],
+      inputPaths: Array[String],
+      broadcastedConf: Broadcast[SerializableConfiguration]): RDD[Row] = {
+    refresh()
```

Comment:
Is that the same with the `refresh` invoking in `DataSourceStrategy`? As every time we build the rdd, we need to refresh the file status.

Actually, I've updated the #8023, by simply removing all of the `refresh` calls in the `DataSourceStrategy`, and the `InsertIntoHadoopFsRelation` will actually update the file status implicitly while writing the data.